text/0097-log-data-model.md

```diff
@@ -0,0 +1,971 @@
+`SeverityNumber` field should be used. `SeverityNumber` can be compared to
+another `SeverityNumber` or to numbers in the 1..24 range (or to the
+corresponding short names).
+
+When severity is used in equality or inequality comparisons (for example in
+filters in the UIs) the recommendation is to attempt to use both `SeverityText`
+and short name of `SeverityNumber` to perform matches. For example if we have a
+record with `SeverityText` field equal to "Informational" and `SeverityNumber`
+field equal to INFO then it may be preferable from the user experience
+perspective to ensure that **severity="Informational"** and **severity="INFO"**
+conditions both to are TRUE for that record.
+
+### Field: `ShortName`
+
+Type: string.
+
+Description: Short event identifier that does not contain varying parts.
+`ShortName` describes what happened (e.g. "ProcessStarted"). Recommended to be
+no longer than 50 characters. Optional. Not guaranteed to be unique in any way.
+Typically used for filtering and grouping purposes in backends.
+
+### Field: `Body`
+
+Type: any.
+
+Description: A value containing the body of the log record (see the description
+of `any` type above). Can be for example a human-readable string message
+(including multi-line) describing the event in a free form or it can be a
+structured data composed of arrays and maps of other values. Can vary for each
+occurrence of the event coming from the same source.
+
+### Field: `Resource`
+
+Type: key/value pair list.
+
+Description: Describes the source of the log, aka
+[resource](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#resources).
+"value" of each pair is of `any` type. Multiple occurrences of events coming
+from the same event source can happen across time and they all have the same
+value of `Resource`. Can contain for example information about the application
+that emits the record or about the infrastructure where the application runs.
+Data formats that represent this data model may be designed in a manner that
+allows the `Resource` field to be recorded only once per batch of log records
+that come from the same source. SHOULD follow OpenTelemetry
+[semantic conventions for Resources](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions).
+
+### Field: `Attributes`
+
+Type: key/value pair list.
+
+Description: Additional information about the specific event occurrence. "value"
+of each pair is of `any` type. Unlike the `Resource` field, which is fixed for a
+particular source, `Attributes` can vary for each occurrence of the event coming
+from the same source. Can contain information about the request context (other
+than TraceId/SpanId). SHOULD follow OpenTelemetry
+[semantic conventions for Attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions).
```

Comment:
I like the minimalness of the semantic conventions linked here. However, in our experience, it comes with significant risk and work to align internally in companies on the rest of the fields that teams will use. Two different people will come up with two different names for the same fields. We have found a lot of value in having a rich schema with optional fields:  https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html

I wanted to mention it here because I think it might be useful as inspiration for extending the semantic conversions.

Reply:
I agree on the importance of having a rich set of standardized field names. ECS is a very good inspiration source. I would even aim to make OpenTelemetry conventions compatible as much as possible with ECS conventions (with possible difference of using a slightly different namespacing approach, e.g. dot notation that is already adopted by OpenTelemetry).

It would be very useful if someone went through both ECS and OpenTelemetry conventions and identified if there are any conflicting definitions between the 2 and then we decide if there is a way to resolve the conflicts.

I think this should be done independently from the data model. As part of that work I would also want want to see a few more things:
- Namespace regulation. A process for adding sub-namespaces that is flexible, fast and easy for the entire industry to contribute, yet ensures there are no name collisions.
- Recommendation for custom attribute naming. How application developers can ensure their custom attributes don't conflict with common attributes, including attributes that will be added to the conventions after the application is released. (reverse domain name notation?).

Reply:
Recorded an issue to work on later: https://github.com/open-telemetry/opentelemetry-specification/issues/583

Resolving this as there is nothing to do as part of this OTEP.